scripts/test-many-configs.sh

```diff
@@ -0,0 +1,345 @@
+        cleanUp test3 ${MODULE}
+        checkNumberRounds test3 12 random-class-method -ge ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log                    #as an input, this function requires the name of folder with test results, expected # rounds, detector type used, and eq/ge (roundsemantics)
+        flakyTestsFound test3 $testCount3 ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+
+        mvn testrunner:testplugin ${rounds}12 ${ogOrderPass}true ${detType}random-class ${verRounds}2 ${PL} &> ${projectDirectory}/test4.log
+        cleanUp test4 ${MODULE}
+        checkDetType test4 random-class ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+        flakyTestsFound test4 $testCount4 ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+
+        mvn testrunner:testplugin ${rounds}8 ${ogOrderPass}true ${detType}random-class-method ${countFirstFail}true ${PL} &> ${projectDirectory}/test5.log
+        cleanUp test5 ${MODULE} #pass in  an extra parameter, the name of the log file, and move it into the new file we create in cleanup func
+        checkNumberRounds test5 8 random-class-method -ge ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+        flakyTestsFound test5 $testCount5 ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+
+        mvn testrunner:testplugin ${rounds}3 ${ogOrderPass}true ${detType}reverse ${PL} &> ${projectDirectory}/test6.log
+        cleanUp test6 ${MODULE}
+        checkNumberRounds test6 1 reverse -ge ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log  
+        flakyTestsFound test6 $testCount6 ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+
+        mvn testrunner:testplugin ${time}120 ${ogOrderPass}true ${detType}reverse-class ${PL} &> ${projectDirectory}/test7.log
+        cleanUp test7 ${MODULE}
+        checkNumberRounds test7 1 reverse-class -ge ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+        flakyTestsFound test7 $testCount7 ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+
+        mvn testrunner:testplugin ${rounds}12 ${semantics}true ${PL} &> ${projectDirectory}/test8.log
+        cleanUp test8 ${MODULE}
+        checkNumberRounds test8 12 random -eq ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+        flakyTestsFound test8 $testCount8 ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+
+        mvn testrunner:testplugin ${rounds}12 ${absPath}/tmp/pathTest ${PL} &> ${projectDirectory}/test9.log
+        checkAbsPath /tmp pathTest ${starr[4]} ${MODULE} |& tee -a ${projectDirectory}/${starr[4]}GeneralLogs.log
+
+    fi
+    cd ${scriptDir}/MC-script-results
+
+done < ${csvFile}
+
+
+
+
+#6. Save all relevant data
+
+if [[ ! -d ARTIFACTS ]]; then
+    mkdir ARTIFACTS
+fi
+for d in test{1,2,3,4,5,6,7,8,9}; do    
```

Comment:
For these test* directories, are they saved right there in the current directory where you are at (which I believe is MC-script-results/)? The following logic seems to be checking that the directory, e.g., test1/, is directly in the current directory, but then to copy it the script tries to $(find -name ${d}), which suggests it is somehow nested underneath somewhere?